docs/2.guide/4.recipes/4.sessions-and-authentication.md

```diff
@@ -0,0 +1,403 @@
+
+    // take the user to the auth-only users index page now that they're logged in
+    await navigateTo("/users");
+
+    // Alternative - Don't use Nuxt Router here so that we can easily trigger a whole page load and get the whole UI refreshed now that the user is logged in. This will perform a full-page load.
+    // window.location.href = "/users";
+  }
+</script>
+
+<template>
+  <div>
+    <form class="space-y-2" @submit.prevent="submitForm">
+      <div>
+        <label for="email" class="block text-sm uppercase">Email</label>
+
+        <input id="email" name="email" v-model="form.email" class="rounded-md px-2 py-1" required />
+      </div>
+
+      <div>
+        <label for="password" class="block text-sm uppercase">password</label>
+
+        <input
+            id="password"
+            name="password"
+            v-model="form.password"
+            class="rounded-md px-2 py-1"
+            type="password"
+            required
+        />
+      </div>
+
+      <div class="pt-2">
+        <button type="submit">Login</button>
+      </div>
+
+      <div v-if="error">{{ error }}</div>
+    </form>
+  </div>
+</template>
+
+```
+
+### 5. Create an API route for login
+
+With the login form created, we need to create an API route to handle the login request. This route should accept a POST request with the email and password in the request body and check the email and password against the database. If the user and password match, we'll set a session cookie to log the user in.
+
+This server API route should be at `/server/api/auth/login.post.ts`. Just like the registration form endpoint, suffixing the filename in `.post.ts.` means that this handler will only respond to post requests.
+
+```typescript [/server/api/auth/login.post.ts]
+import users from "~/database/schema/users";
+import getDatabase from "~/database/database";
+import { eq } from "drizzle-orm";
+import bcrypt from "bcrypt";
+
+export default defineEventHandler(async (event) => {
+  const db = await getDatabase();
+
+  const foundUser = (
+    await db
+      .select({ id: users.id, name: users.name, email: users.email, password: users.password })
+      .from(users)
+      .where(eq(users.email, email))
+      .limit(1)
+  )?.[0];
+
+  // compare the password hash
+  if (!foundUser || !bcrypt.compareSync(password, foundUser.password)) {
+    // return an error if the user is not found or the password doesn't match
+    throw createError({
+      statusCode: 401,
+      statusMessage: "Invalid email or password",
+    });
+  }
+
+  // log in as the selected user
+  await replaceUserSession(event, {
+    user: {
+      id: user.id,
+      name: user.name,
+    },
+    loggedInAt: new Date(),
+  });
+});
+
+
+```
+
+The user should now be logged in! With the session set, we can get the current user session in any API route or page by calling `getUserSession(event)` which is auto-imported as a util function from the `nuxt-auth-utils` package.
```

Comment:
"by calling" can likely be rephrased into "using"
"auto-imported as a util function from the `nuxt-auth-utils` package." can likely be rephrased into "auto-imported from the module", as there's no need to repeat the name of the package too much =)